### PR TITLE
WT-13877 Add place-holder sub-level error codes for customer use cases

### PR DESCRIPTION
**Pull request by DylanSHLiang**
Added all the sub-level error codes listed in the customer use cases. The sub-level error codes are now:
`WT_NONE`
`WT_COMPACTION_ALREADY_RUNNING`
`WT_SESSION_MAX`
`WT_CACHE_OVERFLOW`
`WT_WRITE_CONFLICT`
`WT_OLDEST_FOR_EVICTION`
`WT_CONFLICT_BACKUP`
`WT_CONFLICT_DHANDLE`
`WT_CONFLICT_SCHEMA_LOCK`
`WT_UNCOMMITTED_DATA`
`WT_DIRTY_DATA`
`WT_CONFLICT_TABLE_LOCK`

Additionally extended the python and unit tests to check for all the new sub-level error codes added.
In api_err.py, for the long description of the sub-level errors, there is no long description in the technical design. I've therefore created my own long descriptions for each sub-level error.
I have also added documentation above the sub_errors in api_err.py. The range of integers for sub-level error codes I've written as -32,000 to -32,199 which is the same amount error codes dedicated for normal errors (-31,800 to -31,999).

### CODE REVIEW

File: dist/api_err.py [hunk 1]
```diff
@@ -132,12 +132,61 @@ def __init__(self, name, value, desc, long_desc=None, **flags):
...
+# These numbers cannot change without breaking backward compatibility,
+# and are listed in error value order.
 sub_errors = [
     Error('WT_NONE', -32000,
         'last API call was successful', '''
         This is the default sub-level error code that should be used when there is no
         sub-level error to pair with an error. It indicates that no further context
         exists or is necessary.'''),
+    Error('WT_COMPACTION_ALREADY_RUNNING', -32001,
+        "cannot reconfigure background compaction while it's already running", '''
```
Comment (zunyiliu) [inline dist/api_err.py:147]: @jeremythorp The long descriptions here might not always be accurate. Currently, we lack detailed descriptions for `sub_level_error_code`. Should we discuss and define these error code descriptions and update our tech design doc?

Comment (jeremythorp) [reply dist/api_err.py:147]: hi @zunyiliu , I agree - we should review the error code descriptions and update the tech design doc. 
My suggestion would be to accept the (effectively draft) wording in this PR so that we can merge it and start using these new sub level error codes. Then we add a note in the tech design that the messages need to be reviewed and updated, and also create a ticket (in the project) to revise the text as required once agreement is reached. How does that sound?

Comment (zunyiliu) [reply dist/api_err.py:147]: Sounds good. @jiechenbo I created a ticket [WT-13927](https://jira.mongodb.org/browse/WT-13927) for this, can you schedule that into one of the milestones?

**Review by zunyiliu — APPROVED** [review]
Thanks Dylan, LGTM.

**Review by zunyiliu — CHANGES_REQUESTED** [review]
Revert the approve as discussed and wait for the sub-level error codes short messges changes.

File: test/suite/test_strerror01.py [hunk 2]
```diff
@@ -36,6 +36,24 @@
 # test_strerror01.py
 #     Test generation of sub-level error codes when using calling strerror.
 class test_strerror(wttest.WiredTigerTestCase, suite_subprocess):
+    sub_errors = [
+        (-32000, "WT_NONE: No additional context"),
```
Comment (jiechenbo) [inline test/suite/test_strerror01.py:40]: I would prefer to use the #define error codes themselves. E.g. wiredtiger.WT_NONE

**Review by jiechenbo — APPROVED** [review]
LGTM! Just one comment, let's wait for people to review what they think of the short and long descriptions in the technical design before merging in the PR

File: test/suite/test_strerror01.py (hunk 2 above)
Comment (DylanSHLiang) [reply test/suite/test_strerror01.py]: Fixed

File: dist/api_err.py [hunk 3]
```diff
@@ -132,12 +132,58 @@ def __init__(self, name, value, desc, long_desc=None, **flags):
...
+    Error('WT_CONFLICT_BACKUP', -32006,
+        "Conflict performing operation due to running backup", '''
+        This sub-level error indicates that there is conflict perform the operation
+        because of a running backup in the system.'''),
+    Error('WT_CONFLICT_DHANDLE', -32007,
+        "Another thread currently holds the data handle of the table", '''
+        This sub-level error indicates that a concurrent operation is holding the data
+        handle of the table.'''),
+    Error('WT_CONFLICT_SCHEMA_LOCK', -32008,
+        "Conflict grabbing WiredTiger schema lock", '''
```
Comment (jiechenbo) [inline dist/api_err.py:174]: Just re-reading this, i realised can we reword this to Conflict performing schema operation" instead?

File: dist/api_err.py [hunk 4]
```diff
@@ -139,54 +139,51 @@ def __init__(self, name, value, desc, long_desc=None, **flags):
...
+        "Another thread currently holds the data handle of the table", '''
+        This sub-level error indicates that a concurrent operation is holding the data
+        handle of the table.'''),
     Error('WT_CONFLICT_SCHEMA_LOCK', -32008,
-        "another thread is performing a schema operation", '''
-        This sub-level error is generated when another thread is performing a schema operation,
-        which locks out other operations.'''),
+        "Conflict grabbing WiredTiger schema lock", '''
+        This sub-level error indicates that a concurrent operation is performing a
+        schema operation.'''),
```
[the remark this answers is not in the document]
Comment (jiechenbo) [reply dist/api_err.py:176]: good point! how does this sound instead?
 "This sub-level error indicates that a concurrent operation is performing a schema type operation or currently holds the schema lock."

[the remark this answers is not in the document]
Comment (zunyiliu) [reply dist/api_err.py:176]: Sounds good to me. @DylanSHLiang Can you apply the change?

File: dist/api_err.py (hunk 3 above)
Comment (DylanSHLiang) [reply dist/api_err.py]: Changed in this [commit](https://github.com/wiredtiger/wiredtiger/pull/11349/commits/b71ef6ae4c8697ca7546b18003af0a8989788819)

File: dist/api_err.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (DylanSHLiang) [reply dist/api_err.py]: Changed in this [commit](https://github.com/wiredtiger/wiredtiger/pull/11349/commits/b71ef6ae4c8697ca7546b18003af0a8989788819)

**Review by zunyiliu — APPROVED** [review]
LGTM
